test/libsolidity/semanticTests/various/create_random.sol

```diff
@@ -0,0 +1,39 @@
+contract C {
+    function addr() external returns (address) {
+        return address(this);
+    }
+
+    function testRunner() external returns (address a1, address a2) {
+        assembly {
+            // This is `return(0, 1)`. We are using a simplified/fixed initcode to avoid
+            // instability due to metadata changes.
+            let initcode := hex"60016000f3"
+            mstore(0, initcode)
+
+            a1 := create(0, 0, 5)
+            a2 := create2(0, 0, 5, address())
+        }
+    }
+
+    function testCalc() external returns (address a1, address a2) {
+        a1 = calculateCreate(address(this), 1);
+        a2 = calculateCreate2(address(this), keccak256(hex"60016000f3"), bytes32(uint256(uint160(address(this)))));
+    }
+
+    function calculateCreate(address from, uint256 nonce) private pure returns (address) {
+        assert(nonce <= 127);
+        bytes memory data =
+            bytes.concat(hex"d694", bytes20(uint160(from)), nonce == 0 ? bytes1(hex"80") : bytes1(uint8(nonce)));
```

Comment:
What do you think about allowing `address` in `bytes.concat()` and `string.concat()` without conversion? https://github.com/ethereum/solidity/issues/11593#issuecomment-1109688253

Reply:
Left a comment there.